### PR TITLE
feat: support novelai new api endpoint

### LINKED ISSUE #243: Feature: 支持使用novelai官方的apikey
### Scope

NovelAI

### Describe the problem related to the feature request

novelai自己有官方的apikey，用token不知道为什么没法画

### Describe the solution you'd like

-

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #247: 疑问
我调用了nai3官方api，并根据https://github.com/koishijs/novelai-bot/issues/243进行了修改，
(RTK1X@SW6%6C8Z4S G E4Y
调用中发现，tag短一些还行

但是....一长就出如图示的情况，多次调用也会如此：
image
日志没有记录该错误：
image

### PR DESCRIPTION
**Pull request by MaikoTan**
Based on the investigating from #243 , not tested.

close #243 
close #247

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -209,6 +209,7 @@ export interface Config extends PromptConfig, ParamConfig {
   authLvDefault?: Computed<number>
   output?: Computed<'minimal' | 'default' | 'verbose'>
   features?: FeatureConfig
+  authEndpoint?: string
```
Comment (shigma) [inline src/config.ts:212]: apiEndpoint?

Comment (MaikoTan) [reply src/config.ts:212]: Sure

Comment (MaikoTan) [inline src/config.ts:212]:
```suggestion
  apiEndpoint?: string
```

File: src/config.ts [hunk 2]
```diff
@@ -246,7 +247,8 @@ export const Config = Schema.intersect([
         }),
       ]),
       Schema.object({
-        endpoint: Schema.string().description('API 服务器地址。').default('https://api.novelai.net'),
+        authEndpoint: Schema.string().description('API 服务器地址。').default('https://api.novelai.net'),
```
Comment (MaikoTan) [inline src/config.ts:250]:
```suggestion
        apiEndpoint: Schema.string().description('API 服务器地址。').default('https://api.novelai.net'),
```

File: src/utils.ts [hunk 3]
```diff
@@ -105,13 +105,13 @@ export class NetworkError extends Error {
 
 export async function login(ctx: Context): Promise<string> {
   if (ctx.config.type === 'token') {
-    await ctx.http.get<Subscription>(ctx.config.endpoint + '/user/subscription', {
+    await ctx.http.get<Subscription>(ctx.config.authEndpoint + '/user/subscription', {
```
Comment (MaikoTan) [inline src/utils.ts:108]:
```suggestion
    await ctx.http.get<Subscription>(ctx.config.apiEndpoint + '/user/subscription', {
```

File: src/utils.ts [hunk 4]
```diff
@@ -105,13 +105,13 @@ export class NetworkError extends Error {
...
       timeout: 30000,
       headers: { authorization: 'Bearer ' + ctx.config.token },
     }).catch(NetworkError.catch({ 401: '.invalid-token' }))
     return ctx.config.token
   } else if (ctx.config.type === 'login' && process.env.KOISHI_ENV !== 'browser') {
-    return ctx.http.post(ctx.config.endpoint + '/user/login', {
+    return ctx.http.post(ctx.config.authEndpoint + '/user/login', {
```
Comment (MaikoTan) [inline src/utils.ts:114]:
```suggestion
    return ctx.http.post(ctx.config.apiEndpoint + '/user/login', {
```

**Review by shigma — APPROVED** [review]

**Pull request merged**